core/src/main/java/org/apache/iceberg/util/ManifestFileUtil.java

```diff
@@ -60,7 +60,7 @@ boolean canContain(Object value) {
       }
 
       if (NaNUtil.isNaN(value)) {
-        return containsNaN;
+        return containsNaN == null ? true : containsNaN;
```

Comment:
I'd consider doing this in the constructor to avoid calling the ternary operation in each `canContain`.

```
this.containsNaN = summary.containsNaN() == null || summary.containsNaN();
```

Then this will also be a one line change. I'll leave this up to you, @RussellSpitzer @yyanyy.

Reply:
I'd rather keep it explicit